### PR TITLE
feat: further improvements to the Creating your bot section

### PR DESCRIPTION
**Pull request by monbrey**
This is an attempt to address some common errors seen in support from people who are following the guide (or not following it fully as the case usually is).

This adds an identical tip box to each of the slash command pages that need to be completed in conjunction with each other before the commands will be fully functional (write the commands + load/handle the commands + deploy the commands). People often miss one or more of these steps.

This also makes some changes to the event handling guide to attempt to improve the transition from the other pages. I noticed that our code sample for event handling didn't include the code for command loading. As people don't read and just copy and paste code snippets, it's been added back to the samples to reduce the support load of having to explain it.

### CODE REVIEW

File: guide/creating-your-bot/event-handling.md [hunk 1]
```diff
@@ -31,28 +35,32 @@ client.on(Events.InteractionCreate, async interaction => {
...
+├── node_modules/
 ├── config.json
 ├── deploy-commands.js
 ├── index.js
 ├── package-lock.json
 └── package.json
 ```
 
-Create an `events` folder in the same directory. You can then take your existing events code in `index.js` and move them to `events/ready.js` and `events/interactionCreate.js` files.
+Create an `events` folder in the same directory. You can then move the code from your event listeners in `index.js` to separate files; `events/ready.js` and `events/interactionCreate.js` files.
```
Comment (Jiralite) [inline guide/creating-your-bot/event-handling.md:63]: Shouldn't this be a colon, with "files" at the end omitted?

```suggestion
Create an `events` folder in the same directory. You can then move the code from your event listeners in `index.js` to separate files: `events/ready.js` and `events/interactionCreate.js`.
```

File: code-samples/creating-your-bot/event-handling/index.js [hunk 2]
```diff
@@ -1,10 +1,20 @@
 const fs = require('node:fs');
 const path = require('node:path');
-const { Client, GatewayIntentBits } = require('discord.js');
+const { Client, Collection, Events, GatewayIntentBits } = require('discord.js');
```
Comment (Jiralite) [inline code-samples/creating-your-bot/event-handling/index.js:3]: This is no longer used as the event handler has come into play.

```suggestion
const { Client, Collection, GatewayIntentBits } = require('discord.js');
```

File: guide/creating-your-bot/event-handling.md [hunk 3]
```diff
@@ -94,19 +102,56 @@ module.exports = {
 };
 ```
 :::
+::: code-group-item index.js (after)
+```js
+const fs = require('node:fs');
+const path = require('node:path');
+const { Client, Collection, Events, GatewayIntentBits } = require('discord.js');
```
Comment (Jiralite) [inline guide/creating-your-bot/event-handling.md:109]: This is no longer used as the event handler has come into play.

```suggestion
const { Client, Collection, GatewayIntentBits } = require('discord.js');
```

File: guide/creating-your-bot/event-handling.md [hunk 4]
```diff
@@ -94,19 +102,56 @@ module.exports = {
...
-Next, let's write the code for dynamically retrieving all the event files in the `events` folder. We'll be taking a similar approach to our [command handler](/creating-your-bot/command-handling.md). Place the code below in your `index.js`.
+Next, let's write the code for dynamically retrieving all the event files in the `events` folder. We'll be taking a similar approach to our [command handler](/creating-your-bot/command-handling.md). Place the new code highlighted below in your `index.js`.
 
 `fs.readdirSync().filter()` returns an array of all the file names in the given directory and filters for only `.js` files, i.e. `['ready.js', 'interactionCreate.js']`.
 
-```js {3,5-12}
+```js {18-29}
+const fs = require('node:fs');
+const path = require('node:path');
+const { Client, Collection, Events, GatewayIntentBits } = require('discord.js');
```
Comment (Jiralite) [inline guide/creating-your-bot/event-handling.md:140]: This is no longer used as the event handler has come into play.

```suggestion
const { Client, Collection, GatewayIntentBits } = require('discord.js');
```

File: guide/creating-your-bot/event-handling.md [hunk 5]
```diff
@@ -119,16 +164,20 @@ for (const file of eventFiles) {
...
 
 After this, listening for other events is as easy as creating a new file in the `events` folder. The event handler will automatically retrieve and register it whenever you restart your bot.
 
 ::: tip
-In most cases, you can access your `client` instance in other files by obtaining it from one of the other discord.js structures, e.g. `interaction.client` in the `interactionCreate` event.
+In most cases, you can access your `client` instance in other files by obtaining it from one of the other discord.js structures, e.g. `interaction.client` in the `interactionCreate` event. It doesn't need to manually passed to your events.
```
Comment (Jiralite) [inline guide/creating-your-bot/event-handling.md:180]:
```suggestion
In most cases, you can access your `client` instance in other files by obtaining it from one of the other discord.js structures, e.g. `interaction.client` in the `interactionCreate` event. It doesn't need to be manually passed to your events.
```

**Review by Jiralite — CHANGES_REQUESTED** [review]

**Review by Jiralite — APPROVED** [review]

File: guide/creating-your-bot/command-deployment.md [hunk 6]
```diff
@@ -1,12 +1,16 @@
 # Registering slash commands
 
 ::: tip
-This page assumes you use the same file structure as our [Slash commands](./slash-commands.md) section, and the provided are made to function with that setup. Please carefully read that section first so that you can understand the methods used in this section.
+For fully functional slash commands, there are three important pieces of code that need to be written. They are:
 
-If you already have slash commands set up and deployed for your application and want to learn how to respond to them, refer to the following section on [Command Response Methods](/slash-commands/response-methods.md).
-:::
+1. The [individual command files](slash-commands.html), containing their definitions and functionality.
```
Comment (almostSouji) [inline guide/creating-your-bot/command-deployment.md:6]: I see some irregularities across the guide, but I think we generally omit `.html` on internal links, or if necessary use `.md` https://github.com/discordjs/guide/blob/main/CONTRIBUTING.md#images-and-links

File: guide/creating-your-bot/event-handling.md [hunk 7]
```diff
@@ -119,16 +164,20 @@ for (const file of eventFiles) {
...
+You'll notice the code looks very similar to the command loading above it - read the files in the events folder and load each one individually.
+
+The <DocsLink path="class/Client" /> class in discord.js extends the [`EventEmitter`](https://nodejs.org/api/events.html#events_class_eventemitter) class. Therefore, the `client` object exposes the [`.on()`](https://nodejs.org/api/events.html#events_emitter_on_eventname_listener) and [`.once()`](https://nodejs.org/api/events.html#events_emitter_once_eventname_listener) methods that you can use to register event listeners. These methods take two arguments: the event name and a callback function. These are defined in our separate event files as `name` and `execute`.
```
Comment (almostSouji) [inline guide/creating-your-bot/event-handling.md:173]:
```suggestion
The <DocsLink path="class/Client" /> class in discord.js extends the [`EventEmitter`](https://nodejs.org/api/events.html#events_class_eventemitter) class. Therefore, the `client` object exposes the [`.on()`](https://nodejs.org/api/events.html#events_emitter_on_eventname_listener) and [`.once()`](https://nodejs.org/api/events.html#events_emitter_once_eventname_listener) methods that you can use to register event listeners. These methods take two arguments: the event name and a callback function. These are defined in your separate event files as `name` and `execute`.
```

File: guide/creating-your-bot/event-handling.md [hunk 8]
```diff
@@ -119,16 +164,20 @@ for (const file of eventFiles) {
...
 
 After this, listening for other events is as easy as creating a new file in the `events` folder. The event handler will automatically retrieve and register it whenever you restart your bot.
 
 ::: tip
-In most cases, you can access your `client` instance in other files by obtaining it from one of the other discord.js structures, e.g. `interaction.client` in the `interactionCreate` event.
+In most cases, you can access your `client` instance in other files by obtaining it from one of the other discord.js structures, e.g. `interaction.client` in the `interactionCreate` event. It doesn't need to be manually passed to your events.
```
Comment (almostSouji) [inline guide/creating-your-bot/event-handling.md:180]:
```suggestion
In most cases, you can access your `client` instance in other files by obtaining it from one of the other discord.js structures, e.g. `interaction.client` in the `interactionCreate` event. You do not need to manually pass it to your events.
```
passive -> active

File: guide/creating-your-bot/event-handling.md [hunk 9]
```diff
@@ -119,16 +164,20 @@ for (const file of eventFiles) {
...
+You'll notice the code looks very similar to the command loading above it - read the files in the events folder and load each one individually.
+
+The <DocsLink path="class/Client" /> class in discord.js extends the [`EventEmitter`](https://nodejs.org/api/events.html#events_class_eventemitter) class. Therefore, the `client` object exposes the [`.on()`](https://nodejs.org/api/events.html#events_emitter_on_eventname_listener) and [`.once()`](https://nodejs.org/api/events.html#events_emitter_once_eventname_listener) methods that you can use to register event listeners. These methods take two arguments: the event name and a callback function. These are defined in your separate event files as `name` and `execute`.
```
Comment (Jiralite) [inline guide/creating-your-bot/event-handling.md:173]: I suppose this should be in its correct case:

```suggestion
The <DocsLink path="class/Client" /> class in discord.js extends the [`EventEmitter`](https://nodejs.org/api/events.html#events_class_eventemitter) class. Therefore, the `Client` object exposes the [`.on()`](https://nodejs.org/api/events.html#events_emitter_on_eventname_listener) and [`.once()`](https://nodejs.org/api/events.html#events_emitter_once_eventname_listener) methods that you can use to register event listeners. These methods take two arguments: the event name and a callback function. These are defined in your separate event files as `name` and `execute`.
```

Comment (almostSouji) [reply guide/creating-your-bot/event-handling.md:173]: this is not referring to the type, but to the identifier, right?

**Review by almostSouji — APPROVED** [review]

**Pull request merged**